README.md

```diff
@@ -7,6 +7,85 @@ Version 2.0. See the file "[LICENSE](LICENSE)" for more information.
 
 ## Introduction
 
+mod-settings is a key-value store. It is an attempt to
+make a more secure configuration system than what is
+available today in FOLIO.
+
+See [Fixing the security problem in mod-configuration](https://github.com/MikeTaylor/folio-docs/blob/main/doc/fixing-mod-configuration.md)
+for more information.
+
+It is currently implemented with PostgresQL as storage.
+
+A setting consists of these required properties
+
+* id: unique identifier (UUID)
+* scope: like namespace for the setting. This could be module.
+* key: a readable identifier; not necessarily for display
+* value: a JSON object
+
+And optionally:
+
+* userId: the owner of the setting
+
+We call settings without userId 'global'. global settings must be unique
+for scope and key. Non-global settings must be unique for scope, key
+and userId.
+
+Settings are protected by permissions.
+
+In order to write to a global setting, the client must have permission
+'settings.global.write.'scope, where scope what is used in the setting.
+To read this type of setting, the client must have
+'settings.global.read.'scope.
+
+In order to write to a setting with any userId, which we call a user
+setting, the client must have permission 'settings.users.write.'scope ,
+where scope what is used in the setting. To read this type of setting,
+the client must have 'settings.users.read.'scope.
+
+In order to write to user's own setting, the client must have permission
+'settings.own.write.'scope, where scope what is used in the setting.
+To read this type of setting, the client must have 'settings.own.read.'scope.
+
+With 'read' in this contexts we mean able to read the content of
+the setting. With 'write' in this context we modify storage.
+
+The API is CRUD-like, but with some important changes for some.
+
+Create a setting with:
+
+    POST /config/entries
+
+This is write operation and returns 204 if successful.
+
+Fetch a particular setting with:
+
+    GET /config/entries/{id}
+
+This fetch is protected by permissions. If the client does not
+have permission to read the setting with the scope, it will return
+a 4XX failure.
+
+To get a list settings, use:
+
+    GET /config/entries
+
+The latter takes optional query, limit, offset query parameters.
```

Comment:
Offset is 0-based and defaults to 0

Limit defaults to ???

Query is expressed in CQL and supports queries on the id, scope, key and userId fields. Query terms can mostly only be used in exact-value matching: the exception is that the key field supports left-anchored searches, e.g. `scope=foo and key=bar*` to find all entries in the `foo` scope that begin with `bar`.

Reply:
limit defaults to 10 as RMB had that.

Except that we don't support full text here. I'd really like we do NOT go there.

Reply:
ok. done